lib/matplotlib/_constrained_layout.py

```diff
@@ -87,14 +87,18 @@ def do_constrained_layout(fig, h_pad, w_pad,
        of 0.1 of the figure width between each column.
        If h/wspace < h/w_pad, then the pads are used instead.
 
+    rect : [l, b, w, h]
```

Comment:
Just a minor suggestion for the inputs here.

```suggestion
    rect : tuple of 4 floats
```
This should be the type, you're already specifying what they are below.

Also, do you want to change all the default inputs to tuples instead of a list, `rect=(0, 0, 1, 1)` to avoid mutability concerns?

Reply:
I'm fine to do this, but note that we don't do this elsewhere in the library when we specify rectangles?

Reply:
You're right, it seems rects are specified in a lot of different ways!
```
rect : tuple[float, float, float, float], optional
rect : sequence of float
rect : tuple (left, bottom, right, top), default: (0, 0, 1, 1)
rect : tuple of 4 floats, default: (0, 0, 1, 1)
rect : [left, bottom, width, height]
rect : (float, float, float, float)
```
I'm pretty indifferent on most of those styles, but I don't think the orientation belongs on this first line. My approval still stands regardless of which format of documentation you choose, consolidation of `rect` across matplotlib should be a follow-up issue.

Reply:
I changed this, though this is a private module.